src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -108,38 +108,45 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            
-            for(var i = 0; i < numberOfClients; i++)
+            var exceptions = new ConcurrentQueue<Exception>();
+            Parallel.For(0, numberOfClients, i =>
```

Comment:
@ramonsmits I brought this up in last weeks review. We can definitely improve here. In general @yvesgoeleven is right though you can await the cached task multiple times. @yvesgoeleven @SeanFeldman maybe we can pair on this one today